app/controllers/comments_controller.rb

```diff
@@ -1,15 +1,21 @@
 class CommentsController < ApplicationController
   def create
-    @post_id = params[:post_id]
-    @post = Post.find(params[:post_id])
-    @user = current_user
-    @comments = current_user.comments.create(:post_id => @post_id, :user_id => @user.id, :body => comment_params[:body])
-    if @comments.save
-      flash[:notice] = "Comment was saved."
-      redirect_to url_for(:action=>"show", :controller=>"posts", :format=>"26", :id=>@post.id , :topic_id=>@post.topic_id)
+    if current_user == nil
```

Comment:
Agree. I had actually tried this first, but forgot the '?', and could not figure out why it did not work.